src/ext/clocks.c

```diff
@@ -0,0 +1,44 @@
+#include "clocks.h"
+
+#include <php_config.h>   // for HAVE_CLOCK_GETTIME
+#include <php_version.h>  // for PHP_VERSION_ID
+
+// UNEXPECTED is in different places in PHP 5 and 7
+#if PHP_VERSION_ID < 70000
+#include <Zend/zend.h>
+#else
+#include <Zend/zend_portability.h>
+#endif
+
+#if HAVE_CLOCK_GETTIME
+
+#include <time.h>  // for clock_gettime, CLOCK_MONOTONIC
+
+ddtrace_monotonic_nsec_t ddtrace_monotonic_nsec(void) {
+    struct timespec timespec;
+    if (UNEXPECTED(clock_gettime(CLOCK_MONOTONIC, &timespec))) {
+        return 0;
+    }
+    return ((uint64_t)timespec.tv_sec) * UINT64_C(1000000000) + ((uint64_t)timespec.tv_nsec);
+}
+
+ddtrace_monotonic_usec_t ddtrace_monotonic_usec(void) {
+    struct timespec timespec;
+    if (UNEXPECTED(clock_gettime(CLOCK_MONOTONIC, &timespec))) {
+        return 0;
+    }
+    return ((uint64_t)timespec.tv_sec) * UINT64_C(100000) + ((uint64_t)timespec.tv_nsec / UINT64_C(1000));
```

Comment:
```
Seconds * 100,000 "one hundred thousand" + (nano "billionth of a second" / 1,000 "one thousand")
```

I'm assuming this is for microseconds (µs), which is one millionth of a second, correct? The nano -> micro conversion looks correct, but the seconds -> micro seems like it's missing a zero to make it a million `UINT64_C(1000000)`.

It's times like these that I wish C supported int literal separators. :)

Reply:
Nice catch! Literal separators would definitely help here.